### ISSUE TITLE
Issue with /healthchecks-api

### ISSUE THREAD

**Issue by CarljohanDanling** [Xabaril/AspNetCore.Diagnostics.HealthChecks#327]
Hi. First, thanks for your work with this project!

We are working on an app hosted as an web app in Azure. To get started we have only implemented an url-healthcheck on google's firstpage, just to get some visual feedback from the HealthChecks UI.

The setups looks like this:

```
services.AddHealthChecksUI
(
      setupSettings: setup =>
      {
          setup.AddHealthCheckEndpoint
          (
              name: "Basic health check",
              _hostingEnvironment.IsLocal()
                  ? "https://localhost:5001/healthcheck"
                  : "(hidden URI because of corporate privacy).azurewebsites.net/healthcheck"
          );
      }
);
```

```
services.AddHealthChecks()
     .AddUrlGroup(new Uri("https://www.google.com"),
          name: "Google startpage URL",
          HealthStatus.Healthy);
```

```
app.UseHealthChecks
(
      path: "/healthcheck",
      new HealthCheckOptions
      {
          Predicate = _ => true,
          ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
      });

app.UseHealthChecksUI();
```

Running this on my local machine, valid json response from /healthchecks-api is returned and the UI renders fine at /healthchecks-ui

![health-local-json](https://user-images.githubusercontent.com/42601921/68668741-2cc3c380-0549-11ea-915b-c3d5e12af22c.png)

In Azure the same call to /healthchecks-api throws this error:

![health-azure-error](https://user-images.githubusercontent.com/42601921/68678676-ae721c00-055e-11ea-9fce-d56ab99d03bd.png)

The thing that makes me confused is that the /healthcheck path in the Azure app-url, gives me this: 

![health-azure-json-ok](https://user-images.githubusercontent.com/42601921/68669521-0737b980-054b-11ea-9227-0ef95c5d2278.png)

How can the /healthcheck path receive the correct json when the /healthchecks-api can not?

**Minor investigations:**
Yesterday we turned off the "Authentication / Authorization" on the web app and the call to /healthchecks-api worked great. The thing is we don't want the app to be wide open to public, so we're running with Auth turned on.

**To summarize:** 
My biggest confusion is why - in Azure - /healthcheck can receive correct json reponse but /healthcheck-api can not. Is the api making things "under the hood" that freaks out the Authentication part of Azure? Is the api making external calls somewhere that I can not control?

I would love some help/input on this. Maybe I have setup the healthcheck-code wrong, in that case, please point that out.

**Comment by gustavnyberg** [comment]
Any update on this?

**Comment by CarlosLanderas** [comment]
@CarljohanDanling , what is the implementation of IsLocal?. As it seems to be a custom extension.
Thanks

**Comment by kustic13** [comment]
I've faced a similar issue. Endpoint with health check must be without any checks of user I mean public, in that case, all works fine. I another case the API of healthcheck UI can't pass the required cookies I used to authenticate and authorize users in my web service. The healthcheck UI shows the error on the attached screenshot:
![IMG_20210115_151931_148](https://user-images.githubusercontent.com/32850666/104731926-157ff780-5745-11eb-81c5-f017b3c1d5b4.jpg)

**Comment by dotnetshadow** [comment]
> 
> 
> I've faced a similar issue. Endpoint with health check must be without any checks of user I mean public, in that case, all works fine. I another case the API of healthcheck UI can't pass the required cookies I used to authenticate and authorize users in my web service. The healthcheck UI shows the error on the attached screenshot:
> ![IMG_20210115_151931_148](https://user-images.githubusercontent.com/32850666/104731926-157ff780-5745-11eb-81c5-f017b3c1d5b4.jpg)

Yeah I'm getting the same error, seems the /health endpoints can't be protected or else healthcehck api throws the error
